modules/rostests/apitests/user32/system_menu.c

```diff
@@ -0,0 +1,1149 @@
+/*
+ * Unit tests for window message handling
+ *
+ * Copyright 1999 Ove Kaaven
+ * Copyright 2003 Dimitrie O. Paun
+ * Copyright 2004,2005,2016 Dmitry Timoshkov
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
```

Comment:
Is this a whole copy-paste of the corresponding winetest file, _plus_ your additional testing?
Or is it the _minimal_ needed code from the winetest file in order to test only your additional stuff?

In case it is the former case, I would strongly suggest you remove everything that is not necessary for testing what you want.

Reply:
No, this is not the entire file, I separated only the needed stuff.